BUILD_macos.md

```diff
@@ -78,8 +78,8 @@ PRs with corrections and updates are welcome!
   ```
 
 * To install tools for building the UI:
-  * `brew install node yarn`
-  * Currently, [`yarn`](https://classic.yarnpkg.com/en/docs/install) (< 2.0.0) is required
+  * `brew install node`
+  * `corepack enable`
```

Comment:
Similar to the `corepack enable pnpm` thread (https://github.com/gravitational/teleport/pull/44167#discussion_r1679551343), do you have any opinions/advice on using pnpm to manage the node install?

Ie:

```
brew install pnpm
export PNPM_HOME=~/Library/pnpm
pnpm env use --global 20 # or whatever version
```

- https://pnpm.io/cli/env#use

@ravicious @gzdunek

Reply:
I have never tried this before, but it seems really convenient.
We could even provide a [specific version in `.npmrc`](https://pnpm.io/npmrc#use-node-version), which would be installed automatically for our project when we run any pnpm command:
```
#.npmrc
use-node-version=20.15.0
```

```
grzegorz@mbp teleport % pnpm node -v
Fetching Node.js 20.15.0 ...
v20.15.0
```

But I believe this change should also done for CI to have a single way of installing Node.js and the package manager. However, I think it would be too much for this PR, but we can go back to this idea in the future. 
What do you think?